File: ideal-core/src/main/java/cn/idealframework2/cache/RedisCacheBuilderFactory.java
package cn.idealframework2.cache;

import cn.idealframework2.cache.serialize.ValueSerializer;
import org.springframework.data.redis.core.StringRedisTemplate;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

/**
 * @author 宋志宗 on 2022/9/29
 */
@SuppressWarnings("unused")
public class RedisCacheBuilderFactory {
  @Nullable
  private final String prefix;
  @Nonnull
  private final StringRedisTemplate redisTemplate;

  public RedisCacheBuilderFactory(@Nullable String prefix,
                                  @Nonnull StringRedisTemplate redisTemplate) {
    this.prefix = prefix;
    this.redisTemplate = redisTemplate;
  }

  @Nonnull
  public <K, V> RedisCacheBuilder<K, V> newBuilder(@Nonnull ValueSerializer<V> valueSerializer) {
    return new RedisCacheBuilder<>(prefix, valueSerializer, redisTemplate);
  }
}

File: ideal-core/src/main/kotlin/cn/idealframework2/cache/coroutine/RedisCacheBuilderFactory.kt
package cn.idealframework2.cache.coroutine

import cn.idealframework2.cache.serialize.ValueSerializer
import org.springframework.data.redis.core.ReactiveStringRedisTemplate

/**
 * @author 宋志宗 on 2022/8/15
 */
class RedisCacheBuilderFactory(
  private val prefix: String?,
  private val redisTemplate: ReactiveStringRedisTemplate
) {

  fun <K : Any, V : Any> newBuilder(valueSerializer: ValueSerializer<V>): RedisCacheBuilder<K, V> {
    return RedisCacheBuilder(prefix, valueSerializer, redisTemplate)
  }
}
